source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,343 @@
+``27017``, but you can configure MongoDB to communicate on another port.
+
+.. _node-troubleshooting-connection-firewall:
+
+Configure Your Firewall
+-----------------------
+
+Verify that the ports your MongoDB deployment listens on are not blocked by a
+firewall on the same network. MongoDB uses port ``27017`` by default. To learn
+more about the default ports MongoDB uses and how to change them, see
+:manual:`Default MongoDB Port </reference/default-mongodb-port/>`.
+
+.. warning::
+
+   Do not open a port in your firewall unless you are sure it's the port
+   used by your MongoDB deployment.
+
+ECONNREFUSED Error
+~~~~~~~~~~~~~~~~~~
+
+If the connection is refused when the driver attempts to connect to the MongoDB
+instance, it generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNREFUSED ::<IP address>:<port>
+
+The following sections describe actions you can take to potentially resolve the
+issue.
+
+Start MongoDB in IPv6 Mode
+--------------------------
+
+In Node.js v17 and later, the DNS resolver uses ``IPv6`` by default when both
+the client and host support both. You can configure your MongoDB deployment
+to use ``IPv6`` mode when starting with ``mongod`` or ``mongos``. For more
+information about how to specify ``IPv6`` mode, see
+:manual:`IP Binding </core/security-mongodb-configuration/>` in the server
+manual.
+
+Use IPv4 with Your Mongo Client
+-------------------------------
+
+In Node.js v17 and later, the DNS resolver uses ``IPv6`` by default when both
+the client and host support both. You can explicitly use ``IPv4`` with your
+client by specifying ``family: 4`` as an
+`option to your MongoClient <{+api+}/interfaces/MongoClientOptions.html#family>`__.
+
+.. code-block:: js
+
+   const client = new MongoClient(uri, {
+     family: 4,
+   });
+
+ECONNRESET Error
+~~~~~~~~~~~~~~~~
+
+If the connection is reset when the driver calls ``client.connect()``, it
+generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNRESET ::<IP address>:<port>
+
+The following section describes a method that may help resolve the issue.
+
+Increase the Number of File Descriptors
+---------------------------------------
+
+A file descriptor is a unique identifier associated with an open process. In most
+operating systems, each open connection from the driver is associated with a
+file descriptor. Operating systems typically have a limit on the number of file
+descriptors used by a single process. An ``ECONNRESET`` error can occur
+if the number of connections exceeds this limit.
+
+Consider the following example that creates a client and attempts to connect to
+a local deployment:
+
+.. literalinclude:: /code-snippets/faq/econnresetWithClientConnect-example.js
+  :language: javascript
+  :linenos:
+
+This operation can create a maximum number of 5000 connections as indicated by
```

Comment:
Issue: 
1) I think claiming that the "operation can create... connections" is incorrect. 

2) The example only really shows how to specify maxPoolSize on the connection URI. 


Suggestion:

1) I don't think `connect()` is responsible for creating 5000 connections. I think the `connect()` operation has been a no-op in the Node.js driver for a few years which is why our connection examples need to perform a command to determine whether the connection was successful. 

The driver manages the connection pool. The operation only shows how you specify the setting in the connection string. I would present a scenario without using the code example, and appropriately attribute connection creation to the driver.

2) I think the explanation included in this paragraph is important, but the code snippet does not match the intent of the description. I think either of the following might be better:
- Omitting it 
- Relocating it to after explaining maxPoolSize setting explanation, and using it just to show how to set it on the connection URI.